lib/utils/utils.js

```diff
@@ -228,6 +228,7 @@ function getExternalContractUrl(file) {
   const constants = require('../constants');
   let url;
   const RAW_URL = 'https://raw.githubusercontent.com/';
+  const RAW_URL_IPFS = 'http://localhost:8080/ipfs/'
```

Comment:
It would be better to pass the ipfs url as a parameter. You are right that it comes from `config.storageConfig.upload.getUrl`, but that can change depending on the user's config (though we do set it to `http://localhost:8080/ipfs/` as a default if the user doesn't specify it).

Reply:
What is the best way to retrieve this value from utils.js ?

Reply:
You would need to pass it as a param to the function

Reply:
yes but how can I use `config.` methods inside utils ? Do i need to use an embark object ?

Reply:
No need to use methods. `config` is just a javascript object, so you can just get the url from `config.storageConfig.upload.getUrl`

Reply:
Yes i know, but I have catched an error when I try to import `config.js` into `utils.js`.
`TypeError: utils.filesMatchingPattern is not a function`, from line 361 in `config.json`.Thanks

Reply:
You don't need to import config.js. All of the configs are already loaded in embark's memory. So you need to find the places where `getExternalContractUrl` is used and pass to it the storage config (or more specifically, the uploadURL. Anyway, `getExternalContractUrl` is actually mostly called from config.js, so you have direct access to it from there.

Reply:
I wanted to implement it without pass new parameters to `getExternalContractUrl`, but no problems. I have checked and i should pass `uploadURL` into `getExternalContractURL` from `config.js` and from `file.js`.This parameter would be pass always even that it is useful only for `ipfs` import. How can I do for retrieve `uploadURL` from `file.js`? Thanks in advance

Reply:
Hi @jrainville , any news about this ? If you have time today we can conclude this PR, thanks in advance !!

Reply:
I haven't tried but it seems that File.js is used in the config.js so you should be able to add a new attribute to the constructor and pass self/this

Let me know if that doesn't make sense?

Reply:
Hi @alaibe , I don't sure that I understood. 
Do you mean to add a new attribute to the `file.js` constructor ? I need to use `config.js` into `File.js`.
Thank you very much.